src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
+
+  dead() {
+    const index = Animal.alive.indexOf(this);
```

Comment:
There is no check to see if the animal instance is actually in the 'alive' array before attempting to remove it. It would be safer to only proceed with the 'splice' operation if 'index' is not -1.